### PR TITLE
Cmake imported target

### PR DESCRIPTION
**Pull request by TheOneRing**
@zakalawe @Marcus10110 

See https://github.com/saleae/AnalyzerSDK/pull/9

### CODE REVIEW

**Review by Marcus10110 — CHANGES_REQUESTED** [review]
Thanks!
This builds successfully on Windows, but on MacOS and Linux it does not build - it looks like the AsyncRgbLedAnalyzer's dependancy `Saleae::AnalyzerSDK` is set to NOTFOUND, but I'm not sure why.
Also on Linux TestInstance.cpp is not compiling because it needs `#include <cstring>` but that's an issue with @zakalawe branch, not this.
error message from make:
```
CMakeFiles/AsyncRgbLedAnalyzer.dir/build.make:142: *** target pattern contains no `%’.  Stop.
```
Contents of build.make on that line:
```
libAsyncRgbLedAnalyzer.so: Saleae::AnalyzerSDK-NOTFOUND
```

File: tests/AsyncRgbLedTestDriver.cpp [hunk 1]
```diff
@@ -106,7 +107,7 @@ class LedChannelDataGenerator
               range = (t.max - t.min) * mTolerance;
 
         // we want to offset by a random amount of the range
-        double offsetRandom = drand48() - 0.5; // random is now -0.5 .. 0.5
+        double offsetRandom = rand() - 0.5; // random is now -0.5 .. 0.5
```
Comment (Marcus10110) [inline tests/AsyncRgbLedTestDriver.cpp]: `drand48()` produced a value from [0.0 to 1.0), where `rand()` produces a value from 0 to RAND_MAX.

**Review by Marcus10110 — CHANGES_REQUESTED** [review]
This works on Windows and Linux out of the box, although some of the tests in `AsyncRgbLedTest` fail - that's an implementation problem of the tests though, and unrelated to this PR.
On MacOS it builds, however no tests can run - possibly an rpath problem. When trying to run either test, I get the following error:
```
dyld: Library not loaded: @executable_path/libAnalyzer.dylib
```
I'm not sure what the right way to fix this is. In the past, I've just directly added an rpath to a binary like so:
https://github.com/saleae/Logic/blob/beta/UnitTest/Tests/QtTest/CMakeLists.txt#L311

Lastly, I did see `drand48()` get changed to `rand()`. I've never seen drand48 before, but a quick google search suggests it doesn't return the same range as `rand()`.

**Comment by zakalawe** [comment]
About the rand/drand48 issue, this is my fault, Hannah simply hacked something while she was doing the CMake support. She asked me to fix this with real, portable rand() in the right range and I forgot, I'll fix this later today.

**Comment by TheOneRing** [comment]
@zakalawe the issues should be fixed now?

**Review by Marcus10110 — APPROVED** [review]

**Comment by zakalawe** [comment]
@Marcus10110 do you want me to update this branch? I didn't test it in a while?

**Pull request closed without merging**